web/views/access.py

```diff
@@ -52,33 +56,41 @@ def get_success_url(self, **kwargs):
         return super().get_success_url()
 
 
-@permission_required(Permissions.EDIT, (Dataset, 'pk', 'dataset_pk'))
-def edit_access(request, pk, dataset_pk):
-    access = get_object_or_404(Access, pk=pk)
-    if request.method == 'POST':
-        form = AccessEditForm(request.POST,  request.FILES, instance=access)
-        if form.is_valid():
-            form.save()
-            messages.add_message(request, messages.SUCCESS, "Access definition updated")
-            redirecturl = reverse_lazy('dataset', kwargs={'pk': dataset_pk})
-            return redirect(to=redirecturl, pk=access.id)
-        else:
-            return JsonResponse(
-                {'error':
-                     {'type': 'Edit error', 'messages': [str(e) for e in form.errors]
-                      }}, status=405)
-    else:
-        access.access_notes = None
-        form = AccessEditForm(instance=access)
-
-    log.debug(submit_url=request.get_full_path())
-    return render(request, 'modal_form.html', {'form': form, 'submit_url': request.get_full_path() })
-
-@require_http_methods(["DELETE"])
+class AccessEditView(CheckerMixin, UpdateView, AjaxViewMixin):
+    model = Access
+    template_name = 'accesses/access_form.html'
+    form_class = AccessEditForm
+    permission_required = Permissions.EDIT
```

Comment:
Can we please make a check that only local custodians of the object (dataset) can create/edit the Access object?

Reply:
Or staff

Reply:
So, only data steward and local custodians? Or should it include admins also?

Reply:
@Fancien the ADMIN right  can be kept simple and used just for granting/removing other access rights to other users. 
Lets stick with Local Custodian (as entity level role) and datasteward (as global role).

Reply:
Ok, I added tests to check permissions on objects related to datasets (if more are added in the future, we can add them to the test)
They confirm that only user that are data stewards have Permissions.EDIT on Share, LegalBasis, Documents, DataLocation or Access. Local custodians of a dataset also have these permissions for entities related to the dataset they have custody of.

I also manually tested the website, and users who do not have this permission cannot access the edit page, whether from the modal button (not displayed for them) or from the url (request is rejected)